### ISSUE TITLE
Running subgenerators in a subdirectory

### ISSUE THREAD

**Issue by Sirrah** [yeoman/generator#651]
I would like to have the ability to run a (sub)generator in a subdirectory. In my case I create a `webapp` directory by running (among other options) generator-webapp. The `generator-webapp` should place it's files inside `<project-dir>/webapp`.

With the previous yeoman-generator I could do this by;
1. changing the current working directory to `<project-dir>/webapp`
2. running the subgenerator
3. changing the working directory back to `<project-dir>`

With the current setup for the `runLoop` I no longer have a single before/after point where I can change the working directory appropriately.

An other example would be `generator-mocha`. You can only use this generator if you want your test files in `<project-dir>/test` but you can't use this generator to, for example create a second set of test files.

In other words, you can use the current compose features to create Voltron, or even a  Voltron with one leg, but not a Voltron with two heads and five arms.
